src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -242,8 +242,10 @@ def _find_readme(dirname):
         readme_filename = _find_readme(os.path.dirname(dxapp_json_filename))
         if 'description' in app_spec:
             if readme_filename:
-                logger.warn('"description" field shadows file ' + readme_filename)
-            if not app_spec['description'].strip().endswith('.'):
+                raise dxpy.app_builder.AppBuilderException('Description field found in {file} file! '
+                      'Please transfer content to Readme.md and remove parameter from config file.'.
+                      format(file=dxapp_json_filename))
+            if not app_spec['description'].strip().endswith('.') and not readme_filename:
```

Comment:
You don't need "not readme_filename" here, since you already raised an error above if readme_filename is actually truthy.